### PR TITLE
fix(api): model helpers use targetNames in schemas with CPK enabled

### PR DESCRIPTION
**Pull request by ragingsquirrel3**
Addresses https://github.com/aws-amplify/amplify-flutter/issues/2548

With the latest version of the CLI, new backends have CPK enabled by default. This changes the output of codegen in a way that breaks model helpers. Model helpers look for a field `.targetName` on schema fields in a few places when getting info from related models.

This PR does not add full support for CPK, just keeps model helpers working with new versions of CLI.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_test/lib/test_models/Address.dart [hunk 1]
```diff
@@ -34,183 +35,159 @@ class Address {
...
       return _line1!;
-    } catch (e) {
-      throw AmplifyCodeGenModelException(
-          AmplifyExceptionMessages
-              .codeGenRequiredFieldForceCastExceptionMessage,
-          recoverySuggestion: AmplifyExceptionMessages
-              .codeGenRequiredFieldForceCastRecoverySuggestion,
-          underlyingException: e.toString());
+    } catch(e) {
+      throw new AmplifyCodeGenModelException(
```
Comment (dnys1) [inline packages/amplify_test/lib/test_models/Address.dart:39]: These changes don't seem correct

Comment (HuiSF) [reply packages/amplify_test/lib/test_models/Address.dart:39]: This is what codegen generated I think, we usually reformat these files.
Codegen code definitely needs some clean up...

File: packages/api/amplify_api/lib/src/graphql/graphql_request_factory.dart [hunk 2]
```diff
@@ -207,8 +207,10 @@ class GraphQLRequestFactory {
 
     // e.g. { 'name': { 'eq': 'foo }}
     if (queryPredicate is QueryPredicateOperation) {
+      final association = schema.fields?[queryPredicate.field]?.association;
+      // TODO(ragingsquirrel3): Change key logic when supporting CPK.
       final associatedTargetName =
-          schema.fields?[queryPredicate.field]?.association?.targetName;
+          association?.targetName ?? association?.targetNames?.first;
```
Comment (HuiSF) [inline packages/api/amplify_api/lib/src/graphql/graphql_request_factory.dart:213]: mega nit: can be  `association?.targetNames?.first ?? association?.targetName` as the `targetNames` will be the default, `targetName` is the fallback for backwards compatibility.

**Review by HuiSF — APPROVED** [review]

File: packages/amplify_test/lib/test_models/Address.dart (hunk 1 above)
Comment (ragingsquirrel3) [reply packages/amplify_test/lib/test_models/Address.dart]: I just ran the formatter on all these so hopefully that cleans up the diff.

**Review by HuiSF — APPROVED** [review]

**Review by dnys1 — APPROVED** [review]

**Review by dnys1 — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Pull request merged**